### ISSUE TITLE
💅 [v2.4.7] noUselessStringConcat fix breaks strings concatenated with template functions

### ISSUE THREAD

**Issue by daniele-orlando** [biomejs/biome#9505]
### Environment information

<details>

```bash
CLI:
  Version:                      2.4.7
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v25.8.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.11.0

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

</details>

### Rule name

noUselessStringConcat

### Playground link

https://github.com/user-attachments/files/26020376/Reproduction.zip

### Expected result

`noUselessStringConcat` safe fix should not transform strings concatenated with template functions.

Current rule fix transforms this
```js
t`translate-me` + "!"
```
into this
```js
"translate-me!"
```
which is wrong.

Reproduction:
1) download the [Reproduction.zip](https://github.com/user-attachments/files/26020376/Reproduction.zip)
2) cd ./Reproduction/
3) `npx -p @biomejs/biome@2.4.7 biome lint --only lint/complexity/noUselessStringConcat ./demo.*`
4) check the terminal output and see how Biome suggests (and with --write transforms) a wrong fix.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
